modules/internal/String.chpl

```diff
@@ -233,6 +233,23 @@ module String {
     var _cpindex  : int;
   }
 
+  // Helper routines in support of being able to use ranges of codePointIndex
+  proc chpl_build_bounded_range(low: codePointIndex, high: codePointIndex)
```

Comment:
Do we want to support partially bounded codepoints so you could do something like `str[2:codePointIndex..]`?

Reply:
Hmm, probably so.  I thought of this yesterday but apparently wanted to take a nap too badly to follow through on it.

Reply:
Added in 0238a805b243a1408e5bccb7cc9f458f665429d8